llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -9352,6 +9352,26 @@ SDValue TargetLowering::expandAVG(SDNode *N, SelectionDAG &DAG) const {
     }
   }
 
+  if (Opc == ISD::AVGFLOORU && VT.isScalarInteger() && !isTypeLegal(VT)) {
+    SDValue UAddWithOverflow =
```

Comment:
Please add a brief comment here explaining this expansion, similar to `avgflooru(lhs, rhs) -> add(and(lhs,rhs),lshr(xor(lhs,rhs),1))` below.